WordPress/src/main/java/org/wordpress/android/ui/accounts/login/LoginEmailFragment.java

```diff
@@ -416,9 +427,46 @@ public void onActivityResult(int request, int result, Intent data) {
                         } catch (NullPointerException exception) {
                             disconnectGoogleClient();
                             AppLog.e(T.NUX, "Cannot get ID token from Google sign-in account.", exception);
-                            // TODO: Show error screen.
+                            showErrorDialog(getString(R.string.login_error_generic));
+                        }
+                    } else {
+                        switch (signInResult.getStatus().getStatusCode()) {
+                            // Internal error.
+                            case GoogleSignInStatusCodes.INTERNAL_ERROR:
+                                AppLog.e(T.NUX, "Google Sign-in Failed: internal error.");
+                                showErrorDialog(getString(R.string.login_error_generic)
+                                        + getString(R.string.login_error_suffix_internal));
+                                break;
+                            // Attempted to connect with an invalid account name specified.
+                            case GoogleSignInStatusCodes.INVALID_ACCOUNT:
+                                AppLog.e(T.NUX, "Google Sign-in Failed: invalid account name.");
+                                showErrorDialog(getString(R.string.login_error_generic)
+                                        + getString(R.string.login_error_suffix_invalid));
+                                break;
+                            // Network error.
+                            case GoogleSignInStatusCodes.NETWORK_ERROR:
+                                AppLog.e(T.NUX, "Google Sign-in Failed: network error.");
+                                showErrorDialog(getString(R.string.error_generic_network));
+                                break;
+                            // Cancelled by the user.
+                            case GoogleSignInStatusCodes.SIGN_IN_CANCELLED:
+                                AppLog.e(T.NUX, "Google Sign-in Failed: cancelled by user.");
+                                break;
+                            // Attempt didn't succeed with the current account.
+                            case GoogleSignInStatusCodes.SIGN_IN_FAILED:
+                                AppLog.e(T.NUX, "Google Sign-in Failed: current account failed.");
+                            // Attempted to connect, but the user is not signed in.
+                            case GoogleSignInStatusCodes.SIGN_IN_REQUIRED:
+                                AppLog.e(T.NUX, "Google Sign-in Failed: user is not signed in.");
+                            // Unknown error.
```

Comment:
I'm assuming that the `break` is missing on purpose here. If so, let's add a comment that the fall through is intentional. Thanks!

Reply:
I added the `break` in 0ea92a219ee918a8369285b0fc314c9bc8ab431c to be more explicit.